### DISCUSSION TITLE
Ambient Light

### DISCUSSION THREAD

**Discussion by goldenratio** [jnsmalm/pixi3d#84] [Q&A] [answered]
Hi,
is it possible to achieve AmbientLight like in three.js? https://threejs.org/docs/#api/en/lights/AmbientLight

**Comment by jnsmalm** [comment, marked as answer]
Hey,

Yes you can achieve this by using Image-based lighting. The good thing about this is that you can use an environment as a light source instead of an solid color, which can give you reflections. An example how to use it can be found at https://codesandbox.io/s/github/jnsmalm/pixi3d-sandbox/tree/master/standard-material .You can find additional environments at https://ibl.pixi3d.org/

If you still just want a single color to act as ambient light you can do this as well. Just create a image-based lighting object with single color cube maps.

```javascript
let ibl = new ImageBasedLighting(
    Cubemap.fromColors(new Color(1,0,0)),
    Cubemap.fromColors(new Color(0,0,0))
);
```

The first param is color for diffuse, the second for specular. Experiment to get the look you want.

**Reply by goldenratio** [reply]
thanks for the help!

**Comment by goldenratio** [comment]
for people interested in white light
```
LightingEnvironment.main.imageBasedLighting = new ImageBasedLighting(
      Cubemap.fromColors(new Color(0.8, 0.8, 0.8, 1)),
      Cubemap.fromColors(new Color(0, 0, 0))
    );
```
